File: lib/user_list.rb
# frozen_string_literal: true

require 'discordrb'

# This stores the sprinting role and any users opting in to a sprint,
# and allows them to be mentioned individually
class UserList
  attr_reader :list

  def initialize
    @list = []
  end

  def get_users_sprinting(user)
    list << user
  end

  def user_mentions
    list.map(&:mention).join(' ')
  end
end

File: src/sprints/UserList.ts
interface Mentionable {
  toString(): string
}

export default class UserList {
  list: Mentionable[]

  constructor() {
    this.list = []
  }

  addSprinter(user: Mentionable): void {
    this.list.push(user);
  }

  userMentions(): string {
    return this.list.map(user => user.toString()).join(' ')
  }
}